lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/IMRUDriver.cs

```diff
@@ -446,7 +446,7 @@ public void OnNext(ICompletedTask completedTask)
                 switch (_systemState.CurrentState)
                 {
                     case SystemState.TasksRunning:
-                        _taskManager.RecordCompletedTask(completedTask);
+                        _taskManager.RecordCompletedRunningTask(completedTask);
                         if (_taskManager.AreAllTasksCompleted())
```

Comment:
Why we don't check whether master task has completed here? With the current logic, master task completion doesn't trigger done action, but we wait for either all tasks to complete or for one of tasks/evaluators to fail. If we assume master task completed = success, we should proceed to `DoneAction` immediately after it is recorded